crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1864,6 +1866,23 @@ impl World {
     pub fn run_schedule(&mut self, label: impl AsRef<dyn ScheduleLabel>) {
         self.schedule_scope(label, |world, sched| sched.run(world));
     }
+
+    /// Creates an [`EntityMapper`] from this [`World`] and scoped to the provided [`EntityMap`], then calls the
+    /// provided function with it. This allows one to allocate new entity references in this `World` that are
+    /// guaranteed to never point at a living entity now or in the future. This functionality is useful for safely
+    /// mapping entity identifiers that point at entities outside the source world. The passed function, `f`, is called
+    /// within the scope of this world. Its return value is then returned from `world_scope` as the generic type
+    /// parameter `R`.
+    pub fn world_scope<R>(
```

Comment:
I don't think it should be a part of the world. Can we move it to `EntityMapper` as I suggested in the issue?

Reply:
Yeah that does make more sense. I've made this change, where `world_scope` is now a static method `EntityMapper::world_scope(&mut HashMap<Entity, Entity>, &mut World, /* ... */)`.